### PR TITLE
chore: split dynamodb-mapper-codegen into two modules

### PR DESCRIPTION
**Pull request by lauzadis**
This PR _was_ intended to fix an outstanding FIXME in `Annotations.kt` around switching converterName from a string to ` KClass<ItemConverter<*>>`, but we are blocked by a bug in KSP: https://github.com/google/ksp/issues/1129

Part of addressing that FIXME required splitting dynamodb-mapper-codegen into two modules (ops and schema), which is what I've published here.

## Description of changes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by 0marperez — APPROVED** [review]

File: gradle.properties [hunk 1]
```diff
@@ -14,4 +14,7 @@ smithyKotlinDocBaseUrl=https://sdk.amazonaws.com/kotlin/api/smithy-kotlin/api/$s
 # atomicfu
 kotlinx.atomicfu.enableJvmIrTransformation=true
 # FIXME - https://github.com/Kotlin/kotlinx-atomicfu/issues/274
-kotlinx.atomicfu.enableNativeIrTransformation=false
\ No newline at end of file
+kotlinx.atomicfu.enableNativeIrTransformation=false
+
+# https://github.com/google/ksp/blob/main/docs/ksp2.md
+ksp.useKSP2=true
```
[the remark this answers is not in the document]
Comment (lauzadis) [reply gradle.properties:20]: Nothing major, they have a list of known issues and other migration tips and none of them applied to us. We will need to move to KSP2 once the next version of Kotlin is released:

> KSP1 will not support Kotlin 2.1. Please plan transitioning from KSP1 to KSP2 during Kotlin 2.0.

Comment (ianbotsf) [inline gradle.properties:20]: **Nit**: Please end file with a newline

File: hll/dynamodb-mapper/dynamodb-mapper-ops-codegen/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+*/
+
+description = "DynamoDbMapper ops code generation"
+extra["displayName"] = "AWS :: SDK :: Kotlin :: HLL :: DynamoDbMapper :: Codegen :: Ops"
+extra["moduleName"] = "aws.sdk.kotlin.hll.dynamodbmapper.codegen.ops"
+
+plugins {
+    alias(libs.plugins.kotlin.jvm)
+    `maven-publish`
+}
```
Comment (ianbotsf) [inline hll/dynamodb-mapper/dynamodb-mapper-ops-codegen/build.gradle.kts:13]: **Question**: Is it _necessary_ to publish the ops codegen module? I thought we only had a _build-time_ use case for it.

**Review by ianbotsf — APPROVED** [review]

File: hll/dynamodb-mapper/dynamodb-mapper-ops-codegen/build.gradle.kts (hunk 2 above)
Comment (lauzadis) [reply hll/dynamodb-mapper/dynamodb-mapper-ops-codegen/build.gradle.kts:13]: I think you're right, I'll remove this publication

**Pull request merged**